test/Repository/M4.py

```diff
@@ -36,6 +36,10 @@
 
 test = TestSCons.TestSCons()
 
+# TODO: figure out how to write the coverage data to the locked folder or maybe somewhere else
+if 'COVERAGE_PROCESS_START' in os.environ:
```

Comment:
Can you move the test for running under coverage to QMTest/TestSCons.py such that you can use : if test.coverage_run():   Just to simplify any changes going forward?